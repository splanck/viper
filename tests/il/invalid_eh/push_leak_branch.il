il 0.1.2

func @main(i1 %flag) -> void {
entry:
  eh.push ^handler
  cbr %flag, leak, safe

safe:
  eh.pop
  ret

leak:
  ret

handler ^handler(%err:Error, %tok:ResumeTok):
  eh.entry
  resume.next %tok
}
