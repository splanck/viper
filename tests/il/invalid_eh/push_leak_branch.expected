error: verify.eh.unreleased: main:leak: ret: unmatched eh.push depth 1; path: entry -> leak
